test/e2e/specs/dpop.js

```diff
@@ -0,0 +1,133 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * 
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* global window */
+
+import assert from 'assert';
+import TestApp from '../pageobjects/TestApp';
+import { openPKCE } from '../util/appUtils';
+import { loginRedirect, loginPopup, loginDirect } from '../util/loginUtils';
+
+async function countDPoPKeys () {
+  await browser.setTimeout({ script: 5000 });
+  const result = await browser.executeAsync(function (dbName, storeName, done) {
+    const indexedDB = window.indexedDB;
+    const req = indexedDB.open(dbName, 1);
+
+    req.onsuccess = function () {
+      const db = req.result;
+      const tx = db.transaction(storeName, 'readonly');
+
+      const store = tx.objectStore(storeName);
+
+      const query = store.count();
+      query.onsuccess = function () {
+        done(query.result);
+      };
+
+      tx.oncomplete = function () {
+        db.close();
+      };
+    };
+  }, 'OktaAuthJs', 'DPoPKeys');
+  return result;
+}
+
+async function assertNoRemainingDPoPKeys () {
+  const count = await countDPoPKeys();
+  console.log(count);
+  assert(count === 0);
+}
+
+async function bootstrap(options = {}) {
+  await openPKCE({...options, dpop: true, useInteractionCodeFlow: true});
+  await TestApp.dpopOptionOn.then(el => el.isSelected()).then(isSelected => {
+    assert(isSelected);
+  });
+}
+
+describe('E2E login', () => {
+  const flow = 'pkce';
+
+  it('can login using redirect with responseMode=fragment', async () => {
+    await bootstrap({ responseMode: 'fragment' });
+    await TestApp.responseModeFragment.then(el => el.isSelected()).then(isSelected => {
+      assert(isSelected === true);
+    });
+    await TestApp.interactionCodeOption.then(el => el.click());
+    await loginRedirect('pkce', 'fragment');
+    await TestApp.getUserInfo();
+    await TestApp.assertUserInfo();
+    await TestApp.logoutRedirect();
+    await assertNoRemainingDPoPKeys();
+  });
+
+  // widget-based tests cannot be added until new authjs client is updated in siw
+  // xit('can login using signin widget (no redirect)', async () => {
+  //   await bootstrap();
+  //   await loginWidget(flow);
+  //   await TestApp.getUserInfo();
+  //   await TestApp.assertUserInfo();
+  //   await TestApp.logoutRedirect();
+  //   await assertNoRemainingDPoPKeys();
+  // });
+
+  // // widget-based tests cannot be added until new authjs client is updated in siw
+  // xit('can login using signin widget (with redirect)', async () => {
+  //   let options = { forceRedirect: true };
+  //   await bootstrap(options);
+  //   await loginWidget(flow, true);
+  //   await TestApp.getUserInfo();
+  //   await TestApp.assertUserInfo();
+  //   await TestApp.logoutRedirect();
+  //   await assertNoRemainingDPoPKeys();
+  // });
```

Comment:
Do these belong in `@okta/okta-signin-widget` anyway?

Reply:
perhaps, they already exist in `specs/login`, which is what I based this file on

Reply:
might be best to use the downstream okta-auth-js → okta-signin-widget instead of duplicating?